x-pack/plugins/infra/public/components/asset_details/hooks/use_date_range_provider.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import createContainer from 'constate';
+import { useMemo } from 'react';
+import { parseDateRange } from '../../../utils/datemath';
+import type { AssetDetailsProps } from '../types';
+import { toTimestampRange } from '../utils';
+
+const DEFAULT_DATE_RANGE = {
+  from: 'now-15m',
+  to: 'now',
+};
+
+export type UseAssetDetailsStateProps = Pick<AssetDetailsProps, 'dateRange'>;
+
+export function useDateRangeProvider({ dateRange: rawDateRange }: UseAssetDetailsStateProps) {
```

Comment:
I notice these newer hooks have the word "Provider" as part of the filename and hook name whereas we typically don't include that.  Curious as to why.

Reply:
Good point. Will rename them removing the `provider` suffix